connectors/jetty-connector/src/main/java/org/glassfish/jersey/jetty/connector/JettyConnector.java

```diff
@@ -314,8 +314,14 @@ private Request translateRequest(final ClientRequest clientRequest) {
         request.followRedirects(clientRequest.resolveProperty(ClientProperties.FOLLOW_REDIRECTS, true));
         final Object readTimeout = clientRequest.resolveProperty(ClientProperties.READ_TIMEOUT, -1);
         if (readTimeout != null && readTimeout instanceof Integer && (Integer) readTimeout > 0) {
-            request.timeout((Integer) readTimeout, TimeUnit.MILLISECONDS);
+            request.idleTimeout((Integer) readTimeout, TimeUnit.MILLISECONDS);
```

Comment:
Javadoc does not say much about this, but I think this does not exactly have the same meaning than read time out. This method should also have impact in writing (not only read).

However there is no such method to set a read timeout, and this is more correct than before, so it looks good to me.

Reply:
> Javadoc does not say much about this, but I think this does not exactly have the same meaning than read time out. > This method should also have impact in writing (not only read).

You are right. the timeout may fire when there is a long enough pause during data transfer. I don't know to be honest.
As you say, the situation seems better like this anyway.